Docs diffusion intro (#561)
lumicks/pylake: docs/theory/diffusion/intro.rst

The first hypothesis was that protein diffused through the cytoplasm (in 3D), until they found their target. Berg and von Hippel were the first to demonstrate that the Lac repressor
finds its target faster than can be explained with 3D diffusion alone. 
Now, many experiments have shown that proteins often diffuse both in 1D diffusion (along DNA) and 3D diffusion (in the cytoplasm), also called facilitated diffusion :cite:`Berg1976, Berg1981`, which is more efficient that 3D diffusion alone.  

.. image:: figures/target_search.png

1D diffusion can can be directly visualized by holding a force-stretched DNA between optical tweezers and observing a fluorescently labeled protein bind and move along this DNA tether. 
Pylake has many functions implemented for analyzing data from such an assay. This chapter focusses on the analysis of 1D diffusion of a protein on DNA.

.. image:: figures/1D_diffusion.png

The diffusion coefficient
---------------------------

A purely diffusive protein has the same probability to step to the right as to the left from any location on the DNA, therefore the average displacement is zero. 

> Rather than through velocity, diffusion is therefore quantified through the *diffusion coefficient (D)*. The diffusion coefficient quantifies how fast a protein explores its environment; 

Reviewer: ```suggestion
Since diffusion is distinct from directed motion, which would be characterized by a velocity, it is quantified with a different quantity named the *diffusion coefficient (D)*. The diffusion coefficient quantifies how fast a protein explores its environment; 
```

> the units of the diffusion coefficients are :math:`length^{2}/s`. For 1D diffusion along DNA, D is often expressed in :math:`µm^{2}/s` or :math:`kbp^{2}/s` :cite:`Kochaniak2009`.

Reviewer: Should probably define what you mean by `kbp` in the text?

> Methods for obtaining the diffusion coefficient
> ------------------------------------------------

Reviewer: ```suggestion
Methods for obtaining the diffusion coefficient
-----------------------------------------------
```

> The first step for analyzing a diffusive protein from a kymograph, is to track each binding event. The resuling tracks give the positions of each bound protein over time.

Reviewer: ```suggestion
The first step to analyze a diffusive protein from a kymograph, is to track each binding event. The resulting tracks give the positions of each bound protein over time.
```

These tracks can then be analyzed further to determine the diffusion coefficient. Below we discuss two methods for diffusion analysis that are available in Pylake. 

Mean Squared Displacement analysis
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> A common method for determining the diffusion coefficient, is to first compute the mean squared displacement (MSD) as a function of lag time for each track. 

Reviewer: ```suggestion
A common method for determining the diffusion coefficient is to first compute the mean squared displacement (MSD) as a function of lag time for each track. 
```

For purely diffusive motion, the first part of the MSD curve is linear, with the slope proportional to the diffusion coefficient. For 1D diffusion the MSD is given by: :math:`MSD = 2D\Delta t`.
The diffusion coefficient can be obtained by fitting a linear function to the MSD curve, as illustrated in the figure below (simulated data):

.. image:: figures/MSD_analysis.png

Note that for the MSD curve, the 'lag time' (aka 'time interval') is given on the x-axis, rather than 'time'. 
To compute the MSD for a certain lag time, you scan over a track, one data point at a time and compute the displacement squared 
from that time until that time + the lag time. So each point on the MSD curve, is an average of displacements for a specific lag time, see the :doc:`theory section on diffusive motion</theory/diffusion/diffusion>`

> for the equation for the MSD. 

Reviewer: It might be instructive to mention how this works. That for a track without gaps, you have `N-1` intervals to average over for `lag 1`, `N-2` for `lag 2` etc. To really drive home that you are averaging over time.

The largest possible lag time is the duration of a track and the shortest possible lag is the time resolution, the line time of the kymograph. 
The image above does not show the complete MSD curve. The longest possible lag time would be 2500 seconds, but the MSD curve is only given up to a lag time of 5 seconds. 
The full MSD curve looks as follows:

.. image:: figures/MSD_full.png

> As you can see, the full MSD curve becomes less and less linear for larger lags times. The funky shape of the MSD curve for large lag times is caused by:

Reviewer: Funky? :D

#. The MSD at shorter time lags is averaged over more lags and therefore more accurate

#. Consecutive data points in the MSD are highly correlated

The MSD curve for larger lags is not suitable for a linear fit. An important step in MSD analysis, is therefore to determine which number of lags is optimal for a linear fit and obtaining the diffusion coefficient.

> Taking too many lags, may result in an MSD curve that is not fully linear, while taking too little lags may not give enough data points for a good estimate of the diffusion coefficient.

Reviewer: ```suggestion
Taking too many lags may result in an MSD curve that is not fully linear, while taking too few lags may not give enough data points for a good estimate of the diffusion coefficient.
```

Reviewer: Interestingly, if there is no observation noise and free diffusion, `2` lags is optimal.

Covariance Based Estimator
^^^^^^^^^^^^^^^^^^^^^^^^^^

The Covariance Based Estimator (:ref:`'CVE' <diffusion_cve>`) in Pylake is a method for determining the diffusion coefficient that does not rely on the MSD :cite:`vestergaard2014optimal,vestergaard2016optimizing,Tanasie2022`. 
This method directly computes the diffusion coefficient from the displacements.

Comparing CVE and MSD analysis
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

When tracks are long and diffusion clearly dominates over noise, CVE and MSD analysis perform equally well.
MSD analysis can have a small bias when using very short tracks, or when the diffusion coefficient is very small (when a method is biased, the diffusion coefficients obtained via that method deviates 
from the real diffusion constant). CVE on the other hand, is an unbiased method for determining the diffusion coefficient. 
For a more detailed comparison between CVE and OLS, see the :doc:`theory section on diffusive motion</theory/diffusion/diffusion>`.

> Ensemble Based Estimate
> ^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
Ensemble Based Estimate
^^^^^^^^^^^^^^^^^^^^^^^
```

> The estimate of the diffusion coefficient can further be improved by using an ensemble estimate. 

Reviewer: ```suggestion
The estimate of the diffusion coefficient can be improved further by using an ensemble estimate. 
```

Pylake can compute the average diffusion coefficient of all tracks, where each track is weighted by the number of data points. 
An example of how to compute the ensemble estimate using CVE is given :ref:`here <diffusion_cve>`.

Considerations when tracking a diffusive particle
--------------------------------------------------

Refinement
^^^^^^^^^^
The tracking algorithm in the Pylake kymotracker does not always move from one pixel to the next, but sometimes skips a few pixels within a track. 
In the figure below for example, the track (white) sometimes skips one or multiple pixels.
The missing frames can be added through refinement. Further, refinement can improve the position estimate of the event. The orange line shows the update of the coordinates after refining the white track.

*Note that refining missing frames is necessary for MSD analysis, but not for CVE.* MSD analysis works best when the time between all data points is the same. 

> .. image:: figures/refinement.png

Reviewer: I really like that you mention and show this.

Pylake has :ref:`two methods on refinement<localization_refinement>`:

1) **Centroid refinement**

Centroid refinement always refines missing frames. The disadvantage of this method, is that it does not perform well when tracks are close together or with high background. 
At the moment, this method is the default in the kymotracker.

2) **Gaussian refinement (recommended)**

> Gaussidan refinement fits a Gaussian function to improve the position estimate. This method is better at refining tracks that are close together. 

Reviewer: ```suggestion
Gaussian refinement fits a Gaussian function to improve the position estimate. This method is better at refining tracks that are close together. 
```

Using Gaussian refinement to improve the position estimate of tracks is always recommended. Gaussian refinement can also be used to refine missing frames, but this is not the default. When doing MSD analysis, 

> you have to activate the refinement of missing frames, by setting `refine_missing_frames = True` (see section on :ref:`gaussian refinement<kymotracker_gaussian_refinement>`).

Reviewer: Cross referencing is nice! 👍

> FAQ on diffusion analysis
> --------------------------
> - **The diffusion coefficients for some of the tracks are negative, what does that mean?**

Reviewer: ```suggestion
FAQ on diffusion analysis
-------------------------

- **The diffusion coefficients for some of the tracks are negative, what does that mean?**
```

Reviewer: I really like the FAQ btw.

    Sometimes a diffusion coefficient of a single track is negative. For example, when your diffusion coefficient is low, you will have some diffusion coefficients above and some below zero. 

>     The negative values do not have a biological meaning by themselves, but the average of all the diffusion coefficients from different tracks should still give a good, positively values estimate of the diffusion 

Reviewer: ```suggestion
    The negative values do not have a biological meaning by themselves, but the average of all the diffusion coefficients from different tracks should still give a good, positive-valued estimate of the diffusion 
```

    coefficient (provided that you have enough tracks, and that the protein is diffusing freely). The negative data points *should not* be removed from the dataset, otherwise you get a bias in your estimate. 

- **What happens to the diffusion coefficient if a protein bumps into another protein or against other obstacles?**

>     If there are many diffusive proteins binding to the DNA at once and they are not able to pass each other, the proteins will confine the motion, resulting in sub-diffusive or confined diffusive behavior.

Reviewer: ```suggestion
    If many diffusive proteins bind to the DNA at once and they are not able to pass each other, the proteins will confine the motion, resulting in sub-diffusive or confined diffusive behavior.
```

    If you are interested in free diffusion of a protein, it is best to keep the density of protein binding events low enough, such that the proteins don't meet often. 

>     When the DNA is held between two beads in an optical tweezer experiment, the beads can also confine the motion of a diffusive protein, if the protein diffuses far enough. 

Reviewer: ```suggestion
    When the DNA is held between two beads in an optical tweezer experiment, the beads can also confine the motion of a diffusive protein if the protein diffuses far enough. 
```

    If you want to exclude the effect of the beads, you can consider cutting the track to exclude the part where the protein reaches the bead.

>     Another approach is to account for the presence of the beads when analyzing the data, but this option is not available in Pylake at the moment.

Reviewer: We should really add that feature.

- **How do I account for the motion of the DNA molecule itself?**

>     Though the DNA in an optical tweezer experiment is constrained, it still is fluctuating. The motion of the DNA sets a lower limit on what diffusion coefficient can be observed for a protein diffusing 

Reviewer: ```suggestion
    Though the DNA in an optical tweezer experiment is constrained, it still fluctuates. The motion of the DNA sets a lower limit on what diffusion coefficient can be observed for a protein diffusing 
```
